kernel/kernel-api/src/main/java/io/delta/kernel/internal/TransactionImpl.java

```diff
@@ -301,11 +305,15 @@ private TransactionCommitResult doCommit(
     List<Row> metadataActions = new ArrayList<>();
     metadataActions.add(createCommitInfoSingleAction(attemptCommitInfo.toRow()));
     if (shouldUpdateMetadata || isNewTable) {
-      this.metadata =
-          ColumnMapping.updateColumnMappingMetadata(
-              metadata,
-              ColumnMapping.getColumnMappingMode(metadata.getConfiguration()),
-              isNewTable);
+      /* Update field IDs only if the transaction doesn't include a schema evolution */
+      if (!schemaEvolved) {
+        this.metadata =
+            ColumnMapping.updateColumnMappingMetadata(
+                metadata,
+                ColumnMapping.getColumnMappingMode(metadata.getConfiguration()),
+                isNewTable);
+      }
```

Comment:
@allisonport-db @vkorukanti I renamed `shouldPreserveFieldIds` to `schemaEvolved` which I think makes the distinction that column mapping metadata needs to be updated (new field ID assignment) in the case `shouldUpdateMetadata || isNewTable` **and** it's not a schema evolution.

Reply:
I think this is not needed. I moved the column mapping schema update to txnbuilder in this PR. It simplifies metadata updates in one place. https://github.com/delta-io/delta/pull/4199

Reply:
Oh true, we don't really need to update metadata in the `TransactionImpl`, we could just do it before building the transaction. Good point, I'll clean that up